pkg/csi/service/wcpguest/controller.go

```diff
@@ -420,6 +453,128 @@ func (c *controller) ControllerPublishVolume(ctx context.Context, req *csi.Contr
 	return resp, nil
 }
 
+// controllerPublishForFileVolume is a helper mthod for handling ControllerPublishVolume request for File volumes
+func controllerPublishForFileVolume(ctx context.Context, req *csi.ControllerPublishVolumeRequest, c *controller) (
+	*csi.ControllerPublishVolumeResponse, error) {
+	log := logger.GetLogger(ctx)
+	// Check the feature state for file volume support
+	if !commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.FileVolume) {
+		// Feature is disabled on the cluster
+		return nil, status.Error(codes.InvalidArgument, "File volume not supported.")
+	}
+	// Build the CnsFileAccessConfig instance name and namespace
+	cnsFileAccessConfigInstance := &cnsfileaccessconfigv1alpha1.CnsFileAccessConfig{}
+	cnsFileAccessConfigInstanceName := req.NodeId + "-" + req.VolumeId
+	cnsFileAccessConfigInstanceKey := types.NamespacedName{
+		Namespace: c.supervisorNamespace,
+		Name:      cnsFileAccessConfigInstanceName,
+	}
+	// Check whether the CnsFileAccessConfig instance exist in the supervisor cluster
+	if err := c.cnsOperatorClient.Get(ctx, cnsFileAccessConfigInstanceKey, cnsFileAccessConfigInstance); err != nil {
+		if !apierrors.IsNotFound(err) {
+			// Get() on the CnsFileAccessConfig instance failed with different error
+			msg := fmt.Sprintf("failed to get CnsFileAccessConfig instance: %q. Error: %+v", cnsFileAccessConfigInstance.Name, err)
+			log.Error(msg)
+			return nil, status.Errorf(codes.Internal, msg)
+		}
+		// Create the CnsFileAccessConfig instance since it is not found
+		cnsFileAccessConfigInstance = &cnsfileaccessconfigv1alpha1.CnsFileAccessConfig{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      cnsFileAccessConfigInstanceName,
+				Namespace: c.supervisorNamespace},
+			Spec: cnsfileaccessconfigv1alpha1.CnsFileAccessConfigSpec{
+				VMName:  req.NodeId,
+				PvcName: req.VolumeId,
+			},
+		}
+		log.Debugf("Creating CnsFileAccessConfig instance: %+v", cnsFileAccessConfigInstance)
+		if err := c.cnsOperatorClient.Create(ctx, cnsFileAccessConfigInstance); err != nil {
+			msg := fmt.Sprintf("failed to create cnsFileAccessConfig: %v. Error: %v", cnsFileAccessConfigInstance.Name, err)
+			log.Error(msg)
+			return nil, status.Errorf(codes.Internal, msg)
+		}
+	}
+	log.Debugf("Found CnsFileAccessConfig: %q", cnsFileAccessConfigInstance.Name)
```

Comment:
Can you add the supervisor namespace to log message? Here, and at other places applicable.

Reply:
Done